accessors/spanner/spanner_accessor.go

```diff
@@ -0,0 +1,203 @@
+	}
+	return nil
+}
+
+func GetSpannerLeaderLocation(ctx context.Context, instanceURI string) (string, error) {
+	instanceClient, err := spinstanceadmin.GetOrCreateClient(ctx)
+	if err != nil {
+		return "", err
+	}
+	instanceInfo, err := instanceClient.GetInstance(ctx, &instancepb.GetInstanceRequest{Name: instanceURI})
+	if err != nil {
+		return "", err
+	}
+	instanceConfig, err := instanceClient.GetInstanceConfig(ctx, &instancepb.GetInstanceConfigRequest{Name: instanceInfo.Config})
+	if err != nil {
+		return "", err
+
+	}
+	for _, replica := range instanceConfig.Replicas {
+		if replica.DefaultLeaderLocation {
+			return replica.Location, nil
+		}
+	}
+	return "", fmt.Errorf("no leader found for spanner instance %s while trying fetch location", instanceURI)
+}
+
+func CheckIfChangeStreamExists(ctx context.Context, changeStreamName, dbURI string) (bool, error) {
+	spClient, err := spannerclient.GetOrCreateClient(ctx, dbURI)
+	if err != nil {
+		return false, err
+	}
+	stmt := spanner.Statement{
+		SQL: `SELECT * FROM information_schema.change_streams`,
+	}
+	iter := spClient.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	var cs_catalog, cs_schema, cs_name string
+	var coversAll bool
+	csExists := false
+	for {
+		row, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return false, fmt.Errorf("couldn't read row from change_streams table: %w", err)
+		}
+		err = row.Columns(&cs_catalog, &cs_schema, &cs_name, &coversAll)
+		if err != nil {
+			return false, fmt.Errorf("can't scan row from change_streams table: %v", err)
+		}
+		if cs_name == changeStreamName {
+			csExists = true
+			break
+		}
+	}
+	return csExists, nil
+}
+
+func ValidateChangeStreamOptions(ctx context.Context, changeStreamName, dbURI string) error {
+	spClient, err := spannerclient.GetOrCreateClient(ctx, dbURI)
+	if err != nil {
+		return err
+	}
+	// Validate if change stream options are set correctly.
+	stmt := spanner.Statement{
+		SQL: `SELECT option_value FROM information_schema.change_stream_options
+		WHERE change_stream_name = @p1 AND option_name = 'value_capture_type'`,
+		Params: map[string]interface{}{
+			"p1": changeStreamName,
+		},
+	}
+	iter := spClient.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	var option_value string
+	for {
+		row, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("couldn't read row from change_stream_options table: %w", err)
+		}
+		err = row.Columns(&option_value)
+		if err != nil {
+			return fmt.Errorf("can't scan row from change_stream_options table: %v", err)
+		}
+		if option_value != "NEW_ROW" {
+			return fmt.Errorf("VALUE_CAPTURE_TYPE for changestream %s is not NEW_ROW. Please update the changestream option or create a new one", changeStreamName)
+		}
+	}
+	return nil
+}
+
+func CreateChangeStream(ctx context.Context, changeStreamName, dbURI string) error {
```

Comment:
We should create with retention of 7 days. Default is 1 day which may be too less.